projects/eval-adxl314z/src/examples/dummy/dummy_example.c

```diff
@@ -0,0 +1,175 @@
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "dummy_example.h"
+#include "common_data.h"
+#include "adxl314.h"
+#include "no_os_delay.h"
+#include "no_os_print_log.h"
+#include "no_os_error.h"
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief Dummy example main execution.
+ *
+ * @return ret - Result of the example execution. If working correctly, will
+ *               execute continuously the while(1) loop and will not return.
+*******************************************************************************/
+int dummy_example_main()
+{
+	struct adxl314_dev *adxl314;
+	int ret = -EINVAL;
+
+	/* Clear the screen. */
+	pr_info("%c",27);
+	pr_info("%c",'[');
+	pr_info("%c",'2');
+	pr_info("%c",'J');
+
+	/* Initialize component. */
+	ret = adxl314_init(&adxl314, adxl314_user_init);
+	if (ret)
+		goto error;
+
+	pr_info("ADXL314 successfully initialized.\n");
+
+	/* Perform device self-test. */
+	ret = adxl314_self_test(adxl314);
+	if (ret)
+		goto error;
+
+	/* Set Standby mode, required for performing setup. */
+	ret = adxl314_set_op_mode(adxl314, ADXL314_STDBY);
+	if (ret)
+		goto error;
+
+	/* Set output data rate. */
+	ret = adxl314_set_odr(adxl314, ADXL314_ODR_800HZ);
+	if (ret)
+		goto error;
+
+	/* Set offset for Z axis. */
+	ret = adxl314_set_offset(adxl314, 6, ADXL314_Z_AXIS);
```

Comment:
this example should result in measuring ~9.8 on each axis without any offset

Reply:
The outputs on the three axes without applying any offset are within the values specified by the datasheet. The range of the device is between +/- 200g and that is why the values obtained seem to be far from the actual measured ones.